examples/rust/ex1-volume/src/main.rs

```diff
@@ -0,0 +1,429 @@
+                }
+                3 => {
+                    let q = qdata.len();
+                    qdata.iter_mut().zip(weights.iter()).enumerate().for_each(
+                        |(i, (qdata, weight))| {
+                            *qdata = (jacobian[i + q * 0]
+                                * (jacobian[i + q * 4] * jacobian[i + q * 8]
+                                    - jacobian[i + q * 5] * jacobian[i + q * 7])
+                                - jacobian[i + q * 1]
+                                    * (jacobian[i + q * 3] * jacobian[i + q * 8]
+                                        - jacobian[i + q * 5] * jacobian[i + q * 6])
+                                + jacobian[i + q * 2]
+                                    * (jacobian[i + q * 3] * jacobian[i + q * 7]
+                                        - jacobian[i + q * 4] * jacobian[i + q * 6]))
+                                * weight
+                        },
+                    );
+                }
+                _ => unreachable!(),
+            };
+
+            // Return clean error code
+            0
+        };
+    let qf_build_closure = {
+        let mut qfunction = ceed.q_function_interior(1, Box::new(build_mass));
+        qfunction.add_input("dx", (ncomp_x * dim) as i32, EvalMode::Grad);
+        qfunction.add_input("weights", 1, EvalMode::Weight);
+        qfunction.add_output("qdata", 1, EvalMode::None);
+        qfunction
+    };
+// ---------------------------------------------------------------------------------------
+// ISSUE: EvalMode::Weight is not displaying correctly
+println!("{}", qf_build_closure);
+// ---------------------------------------------------------------------------------------
+    // -- QFunction from gallery
+    let qf_build_named = {
+        let name = format!("Mass{}DBuild", dim);
+        ceed.q_function_interior_by_name(&name)
+    };
+    // -- QFunction for use with Operator
+    let qf_build = if gallery {
+        QFunctionOpt::SomeQFunctionByName(&qf_build_named)
+    } else {
+        QFunctionOpt::SomeQFunction(&qf_build_closure)
+    };
+
+    // Operator that build the quadrature data for the mass operator
+    let mut op_build = ceed.operator(qf_build, QFunctionOpt::None, QFunctionOpt::None);
+    op_build.set_field("dx", &restr_mesh, &basis_mesh, VectorOpt::Active);
+    op_build.set_field(
+        "weights",
+        ElemRestrictionOpt::None,
+        &basis_mesh,
+        VectorOpt::None,
+    );
+    op_build.set_field(
+        "qdata",
+        &restr_qdata,
+        BasisOpt::Collocated,
+        VectorOpt::Active,
+    );
+
+    // Compute the quadrature data for the mass operator
+    let elem_qpts = num_qpts.pow(dim as u32);
+    let mut num_elem = 1;
+    (0..dim).for_each(|d| num_elem *= num_xyz[d]);
+    let mut qdata = ceed.vector(num_elem * elem_qpts);
+    if !test {
+        print!("Computing the quadrature data for the mass operator ...");
+        std::io::stdout().flush().unwrap();
```

Comment:
I'd expect this doesn't need to be flushed if you use `println!` above (because output to a terminal is line-buffered by default).

Reply:
Really, I'm not the biggest fan of this `"Message to the user..."` *basically instant to a human operation* `"done"` thing, so I dropped it